### PR TITLE
Add statuschange event

### PR DESCRIPTION
**Pull request by fson**
Add an event 'statuschange' that has two possible statuses:
* `online`: the session has been established
* `reconnecting`: the session was closed, trying to reconnect

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -83,20 +83,30 @@ function runNgrok(opts, cb) {
...
+		lines.forEach(function (line) {
+			if (!line) return;
+			var data = logfmt.parse(line);
+			if (data.obj === 'web' && data.msg === 'starting web service' && data.addr) {
+				api = request.defaults({
+					baseUrl: 'http://' + data.addr,
+					json: true
+				});
+				cb();
+			} else if (data.obj === 'csess' && data.msg === 'session closed, starting reconnect loop') {
```
Comment (anp) [inline index.js:108]: Should we maybe add tests to make sure that when we update ngrok's binary this doesn't break?

Comment (fson) [reply index.js:108]: Now I've added tests for the `statuschange` events with `starting` and `online` statuses. I don't know how to simulate the `session closed` scenario in the test, so I punted on adding a test for that. Do you maybe have an idea how that could work?

Comment (anp) [reply index.js:108]: I imagine we'd do something horrible like mess with ngrok's socket? I don't think we should worry about that too much though, having tests for a couple of the string parsing cases seems good enough to me!

Comment (fson) [reply index.js:108]: Messing with the socket sounded fun, so I had to give it a quick try 😄 Getting the socket file descriptor from `lsof` was easy enough and I was able to disrupt the connection by closing the socket manually using `lldb`. I wasn't able do this reliably from a script though.

I agree having couple of test cases for output parsing is good enough for now, so I'm going to merge this.

**Pull request merged**